thirdeye-server/src/main/java/ai/startree/thirdeye/core/AppAnalyticsService.java

```diff
@@ -79,4 +100,32 @@ private MonitoredMetricWrapper wrapMonitoredMetric(final AlertMetadataDTO metada
         .setDataset(metadata.getDataset().getName())
         .setDatasource(metadata.getDatasource().getName());
   }
+
+  public ConfusionMatrix computeConfusionMatrixForAnomalies() {
+    int truePositive = 0;
+    int falsePositive = 0;
+    int unclassified;
+    List<MergedAnomalyResultDTO> anomalies = anomalyManager.findAll();
```

Comment:
the query reads all the anomalies, is this on purpose?

anomalies without feedback can be obtained with 
```
count(*)  from merged_anomaly_result_index where anomaly_feedback_id = 0
```
--


relevant feedback ids can be obtained with
```
select anomaly_feedback_id from merged_anomaly_result_index where anomaly_feedback_id != 0
```
--> then you can fetch the feedbacks only, rather than the whole anomalies

Understand it's more work though. @suvodeep-pyne may have an opinion.

Reply:
Exposed new APIs from MergedAnomalyResultManager to accommodate this usecase.